packages/plugins/ITO/src/SiteAdaptor/hooks/useSwapCallback.ts

```diff
@@ -21,13 +25,19 @@ import type { JSON_PayloadInMask } from '../../types.js'
 import { useITO_Contract } from './useITO_Contract.js'
 import { useQualificationContract } from './useQualificationContract.js'
 import { checkAvailability } from '../utils/checkAvailability.js'
+import { Web3 } from '@masknet/web3-providers'
+import { useCallback } from 'react'
+import type { EventLog } from 'web3-core'
 
 export function useSwapCallback(
     payload: JSON_PayloadInMask,
     total: string,
     token: Partial<FungibleToken<ChainId, SchemaType>>,
     isQualificationHasLucky = false,
-) {
+): () => Promise<
+    | { receipt: TransactionReceipt | null; events: { [eventName: string]: EventLog | undefined } | undefined }
+    | undefined
+> {
```

Comment:
The inferred type of 'useClaimCallback' cannot be named without a reference to '../../../../../web3-contracts/node_modules/web3-core/types/index.js'. This is likely not portable. A type annotation is necessary.